src/layout/Pickers.js

```diff
@@ -0,0 +1,59 @@
+import _ from 'lodash/fp'
+import * as F from 'futil-js'
+import React from 'react'
+import { observer } from 'mobx-react'
+import { withStateLens } from '../utils/mobx-react-utils'
+import { value } from '../utils/actout'
+import TextHighlight from './TextHighlight'
+
+export let FilteredPicker = withStateLens({ filter: '' })(
+  observer(
+    ({
+      options,
+      onChange,
+      filter,
+      Input = 'input',
+      Item = 'div',
+      Highlight = TextHighlight,
+    }) => (
+      <div>
+        <Input {...value(filter)} />
+        {_.map(
+          ({ value, label }) => (
+            <Item key={value} onClick={() => onChange(value)}>
+              <Highlight text={label} pattern={F.view(filter)} />
+            </Item>
+          ),
+          _.filter(x => F.matchAllWords(F.view(filter))(x.label), options)
+        )}
```

Comment:
From a performance standpoint, maybe these maps should be separate components that receive just what they need. For example, here, this component will receive:
- `_.filter(x => F.matchAllWords(F.view(filter))(x.label), options)`,
- onChange.

That way, React won't have to re-run the map function if these properties haven't changed.

Reply:
If this idea is interesting to you, there are many other places where we could re-think small things this way.

Reply:
We don't know that's the case, and if so, we don't know what the performance impact will be. That's something to follow up on afterwards and shouldn't block 1.0